themes/default/content/blog/mlops-the-ai-challenge-is-cloud-not-code/index.md

```diff
@@ -21,18 +21,18 @@ tags:
-
-This high-level diagram easily illustrates the complexity involved.
+However, using AI/ML in your own products has some major challenges and obstacles. Below is a diagram of the end to end workflow of building and using an AI model:  preparing the data, training a model, fine-tuning a model, hosting and running a model, building a backend service to serve with the model, and building the user interface to that interacts with the model. Most AI engineers are only involved in a few steps of the process. However, there is one challenge that is common across the entire workflow:  creating and managing the cloud infrastructure is hard. 
 
 > Figure 1. A sneak peak at the map of ML development and product path
 ![A sneak peak at the map of ML development and product path](image.png)
 
-Intended to help users blaze a trail of success, this blog post is the first installment of the *Pulumi Python + MLOps* series. Through the journey of this series we will start from the beginning with a basic app and the simple cloud deployment architecture in this post, and follow the chatbot development through its evolution into a robust and resilient cloud native service.
+Training a model requires spinning up hyper scale GPU clusters. Preparing and consuming the training or fine-tuning data requires managing data warehouses and buckets of object storage. Building an AI backend service requires packaging the model, deploying and scaling the model across compute infrastructure, creating and managing the network infrastructure like load balancing, and managing the vector databases. Building great frontend user experiences requires provisioning and managing CDNs and web application services. Suffice to say:  *The most difficult challenge that we face with AI today is a problem of cloud orchestration*.
 
-> # *The most difficult challenge that we face with AI today is a problem of cloud orchestration*
+Most of this cloud orchestration today is done manually or with complex scripts. This is usually frought with errors and simply doesn't scale. Instead, most of your time is spent making cloud infrastructure work instead of the hard part of AI. Infrastructure as code (IaC) is a key component to solving the cloud orchestration challenge present in AI/ML. By applying the rigor and precision of software development practices to cloud operations, IaC offers AI/ML professionals an efficient, reliable, and predictable way to develop at the highest velocity. Pulumi is an open source infrastructure as code platform that allows engineers to use any programming languages, including Python, to provision and manage AI infrastructure. 
```

Comment:
```suggestion
Most of this cloud orchestration today is done manually or with complex scripts. This is usually fraught with errors and simply doesn't scale. Too much time is spent making cloud infrastructure work instead of focusing on actual AI/ML tasks. Infrastructure as code (IaC) is a key component to solving the cloud orchestration challenge present in AI/ML. By applying the rigor and precision of software development practices to cloud operations, IaC offers AI/ML professionals an efficient, reliable, and predictable way to develop at the highest velocity. Pulumi is an open source infrastructure as code platform that allows engineers to use any programming languages, including Python, to provision and manage AI infrastructure. 
```